cumulusci/tasks/salesforce/users/tests/test_permsets.py

```diff
@@ -199,3 +203,456 @@ def test_create_permset_raises(self):
+        assert "0PG000000000001" in responses.calls[2].request.body
+
+    @responses.activate
+    def test_create_permsetgroup__alias_raises(self):
+        task = create_task(
+            AssignPermissionSetGroups,
+            {
+                "api_names": "PermSetGroup1,PermSetGroup2",
+                "user_alias": "test",
+            },
+        )
+
+        responses.add(
+            method="GET",
+            url=f"{task.org_config.instance_url}/services/data/v50.0/query/?q=SELECT+Id%2C%0A++++++++++++++++++++++++++++%28SELECT+PermissionSetGroupId%0A+++++++++++++++++++++++++++++FROM+PermissionSetAssignments%29%0A++++++++++++++++++++++++FROM+User%0A++++++++++++++++++++++++WHERE+Alias+%3D+%27test%27",
+            status=200,
+            json={
+                "done": True,
+                "totalSize": 0,
+                "records": [],
+            },
+        )
+        with pytest.raises(CumulusCIException):
+            task()
+
+    @responses.activate
+    def test_create_permsetgroup_raises(self):
+        task = create_task(
+            AssignPermissionSetGroups,
+            {
+                "api_names": "PermSetGroup1,PermSetGroup2,PermSetGroup3",
+            },
+        )
+
+        responses.add(
+            method="GET",
+            url=f"{task.org_config.instance_url}/services/data/v50.0/query/?q=SELECT+Id%2C%0A++++++++++++++++++++++++++++%28SELECT+PermissionSetGroupId%0A+++++++++++++++++++++++++++++FROM+PermissionSetAssignments%29%0A++++++++++++++++++++++++FROM+User%0A++++++++++++++++++++++++WHERE+Username+%3D+%27test-cci%40example.com%27",
+            status=200,
+            json={
+                "done": True,
+                "totalSize": 1,
+                "records": [
+                    {
+                        "Id": "005000000000000",
+                        "PermissionSetAssignments": {
+                            "done": True,
+                            "totalSize": 1,
+                            "records": [{"PermissionSetGroupId": "0PG000000000000"}],
+                        },
+                    }
+                ],
+            },
+        )
+        responses.add(
+            method="GET",
+            url=f"{task.org_config.instance_url}/services/data/v50.0/query/?q=SELECT+Id%2C+DeveloperName+FROM+PermissionSetGroup+WHERE+DeveloperName+IN+%28%27PermSetGroup1%27%2C+%27PermSetGroup2%27%2C+%27PermSetGroup3%27%29",
+            status=200,
+            json={
+                "done": True,
+                "totalSize": 1,
+                "records": [
+                    {
+                        "Id": "0PG000000000000",
+                        "DeveloperName": "PermSetGroup1",
+                    },
+                    {
+                        "Id": "0PG000000000001",
+                        "DeveloperName": "PermSetGroup2",
+                    },
+                ],
+            },
+        )
+
+        with pytest.raises(CumulusCIException):
+            task()
```

Comment:
I appreciate the verbose tests that don't reuse code. Makes it easy to confirm that the intended behavior makes sense.